SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java

```diff
@@ -128,8 +128,7 @@ private boolean isRecoveryStatusUpdate(Optional<SingularityTaskStatusHolder> pre
     if (!previousTaskStatusHolder.isPresent() // Task was already removed from the active list
         && !taskState.isDone()
         && newTaskStatusHolder.getTaskStatus().isPresent()
-        && ACTIVE_STATES.contains(newTaskStatusHolder.getTaskStatus().get().getState())
-        && reason == Reason.REASON_AGENT_REREGISTERED) {
```

Comment:
Removing this additional condition will let us process and reconcile all tasks that are in mesos master but not zk, not only the cones with an explicit reconciliation update state